testdata/go/v3/memcached-operator/test/e2e/e2e_test.go

```diff
@@ -0,0 +1,175 @@
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	//nolint:golint
+	//nolint:revive
+	. "github.com/onsi/ginkgo"
+
+	//nolint:golint
+	//nolint:revive
+	. "github.com/onsi/gomega"
+
+	"github.com/example/memcached-operator/test/utils"
+)
+
+// namespace store the ns where the Operator and Operand will be executed
+const namespace = "memcached-operator-system"
+
+var _ = Describe("memcached", func() {
+
+	Context("ensure that Operator and Operand(s) can run in restricted namespaces", func() {
+		BeforeEach(func() {
+			By("installing prometheus operator")
+			Expect(utils.InstallPrometheusOperator()).To(Succeed())
+
+			By("installing the cert-manager")
+			Expect(utils.InstallCertManager()).To(Succeed())
+
+			By("creating manager namespace")
+			cmd := exec.Command("kubectl", "create", "ns", namespace)
+			_, _ = utils.Run(cmd)
+
+			By("labeling all namespaces to warn when we apply the manifest if would violate the PodStandards")
+			cmd = exec.Command("kubectl", "label", "--overwrite", "ns", "--all",
+				"pod-security.kubernetes.io/audit=restricted",
+				"pod-security.kubernetes.io/enforce-version=v1.24",
+				"pod-security.kubernetes.io/warn=restricted")
+			_, err := utils.Run(cmd)
+			ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+			By("labeling enforce the namespace where the Operator and Operand(s) will run")
+			cmd = exec.Command("kubectl", "label", "--overwrite", "ns", namespace,
+				"pod-security.kubernetes.io/audit=restricted",
+				"pod-security.kubernetes.io/enforce-version=v1.24",
+				"pod-security.kubernetes.io/enforce=restricted")
+			_, err = utils.Run(cmd)
+			Expect(err).To(Not(HaveOccurred()))
+		})
+
+		AfterEach(func() {
+			By("uninstalling the Prometheus manager bundle")
+			utils.UninstallPrometheusOperator()
+
+			By("uninstalling the cert-manager bundle")
+			utils.UninstallCertManager()
+
+			By("removing manager namespace")
+			cmd := exec.Command("kubectl", "create", "ns", namespace)
+			_, _ = utils.Run(cmd)
+		})
+
+		It("should successfully run the Memcached Operator", func() {
+			var controllerPodName string
+			var err error
+			projectDir, _ := utils.GetProjectDir()
+
+			// operatorImage store the name of the imahe used in the example
+			const operatorImage = "example.com/memcached-operator:v0.0.1"
+
+			By("building the manager(Operator) image")
+			cmd := exec.Command("make", "docker-build", "IMG=example.com/memcached-operator:v0.0.1")
+			_, err = utils.Run(cmd)
+			ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+			By("loading the the manager(Operator) image on Kind")
+			err = utils.LoadImageToKindClusterWithName("example.com/memcached-operator:v0.0.1")
+			ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+			By("pulling Operand image")
+			cmd = exec.Command("docker", "pull", "memcached:1.4.36-alpine")
+			_, err = utils.Run(cmd)
+			ExpectWithOffset(1, err).NotTo(HaveOccurred())
```

Comment:
Is this step necessary? This looks instead like it is pulling the Operand image to the host machine but not adding it to the KinD cluster

Reply:
we need to do both right? pull and load

Reply:
good catcher